src/ai/defaultai.cc

```diff
@@ -6237,14 +6406,18 @@ void DefaultAI::consider_productionsite_influence(BuildableField& field,
 	for (const Widelands::DescriptionIndex& i : bo.ware_outputs) {
 		++field.producers_nearby.at(i);
 	}
-	if (bo.has_collected_map_resource()) {
-		++field.collecting_producers_nearby.at(bo.get_collected_map_resource());
+	for (const auto& supported_building : bo.supported_producers) {
+		// Using std::map syntax to create entry on demand.
+		++field.supported_producers_nearby[supported_building.first];
```

Comment:
I believe this is not doing what it should according to the variable name. 
As far as I understand this we are adding the supported buildings of the building we found here, while we should add the building itself as the goal is to identifiy the number of exisitng dedicated buildings in the vicinity of a single field. So we need to count thze building we found rather then the ones that are supported by our building found.